pkg/concurrency/keyed_mutex_test.go

```diff
@@ -25,7 +25,7 @@ func TestKeyedMutexSameKey(t *testing.T) {
 	}()
 
 	a.False(signal.IsDone())
-	km.Unlock(testKey)
+	func() { defer km.Unlock(testKey) }() // suppress the error for calling Unlock()
```

Comment:
I was trying to find that error that `km.Unlock(testKey)` returns, but there is none :)

```suggestion
	func() { defer km.Unlock(testKey) }() // Suppress linter error for calling Unlock()
```

Reply:
I wonder how it's working? Do we need defer in this function as it's the only line?

Reply:
```
	signal := NewSignal()
	km.DoWithLock(testKey, func() {
		go km.DoWithLock(testKey, func() {
			a.True(signal.Signal())
		})

		a.False(signal.IsDone())
	})
	a.True(WaitWithTimeout(signal.WaitC(), 5*time.Second))
	```

Reply:
```go
func (k *KeyedMutex) Lock(key string) 
func (k *KeyedMutex) Unlock(key string) 
```
are only used in tests how about removing them or unexporting?

Reply:
@vikin91 technically, it's not a linter error but a roxvet analyzer one, which is somewhat too long for the comment. I'll see what I can do.

@janisz not sure about line skipping, likely a lot of work with diminishing return. It works in the way that the call is now deferred and is hence allowed. Also not sure I understand what you suggest in 

> are only used in tests how about removing them or unexporting?

because the functions are used extensively in datastore impls. If you mean switching to the `DoWithLock()` pattern only, I'm afraid this will be too intrusive.

Reply:
Updated.

Reply:
>  I'm afraid this will be too intrusive.

How about marking Lock/Unlock deprecated with info to use WithLock

Reply:
This won't going to fly because `defer ....Unlock()` is a common, valid pattern that is loved by folks: https://redhat-internal.slack.com/archives/C02CFR7GQUS/p1702571262753899

Reply:
Haven't heard anything for some time, will now resolve and proceed. If there are still concerns, let me know.